### PR TITLE
Add a PhoneNumber class to handle formatting and actions on numbers

### PR DESCRIPTION
**Pull request by mikeheidenreich**
We value collaboration at Invoca, and focus on peer mentorship as a way for the whole team to learn and thrive. As part of this process, we conduct code reviews during our development cycle. In this phase of the interview, we would like you to review and comment on this Pull Request. You can assume that this Pull Request was written by an intern, who is new to professional software development.  Please take about 20-30 minutes to review this Pull Request and comment as if you were reviewing an intern's code.

### CODE REVIEW

File: phone_number.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    "(#{find_area_code}) #{find_exchange}-#{find_line_number}"
+  end
+
+  def find_area_code
+    open_paren = @number.index('(')
+    close_paren = @number.index(')')
+    if open_paren
+      if close_paren
+        @number[(open_paren + 1)..(close_paren - 1)]
+      end
```
Comment (tomy) [inline phone_number.rb:46]: This will break if someone enters a number (1234) 56-7890. The else section would still work in this case. Best to strip out all punctuation, make sure there are ten digits and take the first three. That works in cases where the number is entered 123-456-7890 and 123.456.7890.

**Review by tomy — COMMENTED** [review]
The tests look good. I might the odd edge case of someone passing in parens incorrectly (1234) 56-7890.
